code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/InstancePurgePostListener.java

```diff
@@ -0,0 +1,63 @@
+package io.cattle.platform.servicediscovery.process;
```

Comment:
Any reason why we do deallocation on Purge, not Remove? Also when allocation happens - on create or on Start? If on Start, then the deallocation should happen on Stop.